pkg/virt-api/rest/vnc.go

```diff
@@ -34,5 +34,8 @@ func validateVMIForVNC(vmi *v1.VirtualMachineInstance) *errors.StatusError {
 		log.Log.Object(vmi).Reason(err).Error("Can't establish VNC connection.")
 		return errors.NewBadRequest(err.Error())
 	}
+	if !vmi.IsRunning() {
+		return errors.NewBadRequest(vmiNotRunning)
+	}
```

Comment:
> If someone tries to connect to a paused VMI's console currently, what happens?
You will see
`Successfully connected to sle15sp4-minimal console. The escape sequence is ^]`
but it will not accept any commands.

Reply:
In the PR you referred, IIUC, we're talking about an automated process revolving a console connection to a VMI that fails because of timeout. This is however is different than a user (non-automated) experience.

For example, as a user, isn't it reasonable to start a VM then right away try to connect to its console? Would users prefer to wait for a few seconds until the VM starts then having an error popped up?

Reply:
> For example, as a user, isn't it reasonable to start a VM then right away try to connect to its console? Would users prefer to wait for a few seconds until the VM starts then having an error popped up?

I think currently with KubeVirt you cannot connect to the console untill a VM is in `Running` state. Just to note: this does not change the existing behaviour. I simply had to move this check as it previously was in 

https://github.com/kubevirt/kubevirt/blob/a3aaf1ace40c44c9dafaf2c091851d5cdb829fd7/pkg/virt-api/rest/subresource.go#L172-L177

In this PR I changed the condition: 

```go
func (app *SubresourceAPIApp) getVirtHandlerConnForVMI(vmi *v1.VirtualMachineInstance) (kubecli.VirtHandlerConn, error) {
	if !vmi.IsRunning() && !vmi.IsScheduled() {
		return nil, goerror.New(fmt.Sprintf("Unable to connect to VirtualMachineInstance because phase is %s instead of %s or %s", vmi.Status.Phase, v1.Running, v1.Scheduled))
	}
	return kubecli.NewVirtHandlerClient(app.virtCli).Port(app.consoleServerPort).ForNode(vmi.Status.NodeName), nil
}
```
Because the certificates should be fetched before a VM is in `Running`. But then I had to move this `IsRunning` check specifically to vnc and console